### PR TITLE
remove camp_DIR export

### PR DESCRIPTION
**Pull request by trws**
# Summary

Test removing the camp_DIR export.  The existing camp paths should also be searched by the default logic, so this should work with current camp even without updating them.  I will make a PR over there to update them so it looks more consistent anyway, but think we should test this.

I have *not* tested this with external dependencies very well, it's possible the config file is partially broken, so we need some testing downstream before this can go in.

### CODE REVIEW

File: share/raja/cmake/RAJA-config.cmake.in [hunk 1]
```diff
@@ -30,21 +30,14 @@
...
-    if (EXISTS @EXTERNAL_CAMP_INSTALL_DIR@)
-      set(camp_DIR @EXTERNAL_CAMP_INSTALL_DIR@)
-    else ()
-      set(camp_DIR @CMAKE_INSTALL_PREFIX@/lib/cmake/camp)
-    endif ()
-  endif ()
-  find_package(camp REQUIRED PATHS ${camp_DIR} NO_DEFAULT_PATH)
-  include(@CMAKE_INSTALL_PREFIX@/share/raja/cmake/RAJA.cmake)
-endif()
+find_dependency(camp REQUIRED PATHS "@CMAKE_INSTALL_PREFIX@")
```
[the remark this answers is not in the document]
Comment (trws) [reply share/raja/cmake/RAJA-config.cmake.in:35]:
```suggestion
find_dependency(camp REQUIRED PATHS "@PACKAGE_INSTALL_PREFIX@")
```

**Comment by rhornung67** [comment]
@davidbeckingsale if you are good with this PR, please approve.

**Comment by davidbeckingsale** [comment]
> @davidbeckingsale if you are good with this PR, please approve.

I think this needs some tweaks. I'm working on an Umpire equivalent here: https://github.com/LLNL/Umpire/pull/669

File: share/raja/cmake/RAJA-config.cmake.in [hunk 2]
```diff
@@ -30,21 +30,13 @@
...
-  find_package(camp REQUIRED PATHS ${camp_DIR} NO_DEFAULT_PATH)
-  include(@CMAKE_INSTALL_PREFIX@/share/raja/cmake/RAJA.cmake)
-endif()
+if (NOT TARGET camp)
+  include(CMakeFindDependencyMacro)
+  find_dependency(camp REQUIRED PATHS "@PACKAGE_CMAKE_INSTALL_PREFIX@")
+endif ()
 
-# Export version number
-set(RAJA_VERSION_MAJOR @RAJA_VERSION_MAJOR@)
```
Comment (davidbeckingsale) [inline share/raja/cmake/RAJA-config.cmake.in:48]: I removed the version stuff here and used `write_basic_package_version_file`

**Comment by davidbeckingsale** [comment]
@trws this matches what I added to Umpire, and I tested it out with a small local example.

File: share/raja/cmake/RAJA-config.cmake.in (hunk 2 above)
Comment (rhornung67) [reply share/raja/cmake/RAJA-config.cmake.in:51]: Will these macros still get set? https://github.com/LLNL/RAJA/blob/develop/include/RAJA/config.hpp.in#L274

Projects use these to check our version information.

Comment (trws) [reply share/raja/cmake/RAJA-config.cmake.in:51]: I think these now come in from the generated RAJA.cmake don't they?

Comment (davidbeckingsale) [reply share/raja/cmake/RAJA-config.cmake.in:51]: Yup, the macros are still set - they get generated at config time. This change lets a project put the version number in the `find_package(RAJA)` statement in their own CMake, if they care to do that.

Comment (davidbeckingsale) [reply share/raja/cmake/RAJA-config.cmake.in:51]: @trws @rhornung67 - I think that these variables will actually be set, based on the following, from https://cmake.org/cmake/help/latest/command/find_package.html#config-mode-version-selection:

```
These variables are checked by the find_package command to determine whether the configuration file provides an acceptable version. They are not available after the find_package call returns. If the version is acceptable the following variables are set:

<PackageName>_VERSION

    Full provided version string
<PackageName>_VERSION_MAJOR

    Major version if provided, else 0
<PackageName>_VERSION_MINOR

    Minor version if provided, else 0
<PackageName>_VERSION_PATCH

    Patch version if provided, else 0
```

**Comment by trws** [comment]
Oh nice, yeah they will.  As long as its set appropriately in the 
project call anyway.

I’m actually going through to fix how camp is doing this, I don’t 
really have it set up to consistently present versions that are easy to 
deal with right now.

**Comment by rhornung67** [comment]
@trws @davidbeckingsale changes to the RAJA cmake stuff in this PR is causing all CI jobs to fail with the error:

 CMake Error at cmake/SetupRajaConfig.cmake:74 (configure_package_config_file):
  Unknown CMake command "configure_package_config_file".

Do we need to use a newer CMake version or add something like include(CMakePackageConfigHelpers) to the top-level CMakeLists.txt file?

**Comment by trws** [comment]
That functionality has been around for a long, long time.  Maybe it isn't included by default in some older versions or something? The strange part is that it works some places...  Either way, the include should take care of it AFAICT.

File: cmake/SetupRajaConfig.cmake [hunk 3]
```diff
@@ -71,11 +71,23 @@ configure_file(${PROJECT_SOURCE_DIR}/include/RAJA/config.hpp.in
   ${PROJECT_BINARY_DIR}/include/RAJA/config.hpp)
 
 # Configure CMake config
-configure_file(${PROJECT_SOURCE_DIR}/share/raja/cmake/RAJA-config.cmake.in
-  ${PROJECT_BINARY_DIR}/share/raja/cmake/raja-config.cmake @ONLY)
+configure_package_config_file(
```
Comment (trws) [inline cmake/SetupRajaConfig.cmake:74]:
```suggestion
include(CMakePackageConfigHelpers)
configure_package_config_file(
```

**Comment by trws** [comment]
Threw that in through a suggestion just before it gets used, hopefully this will move things along.

Anyone else seeing crazy long times on the Azure pipelines this morning?

**Comment by rhornung67** [comment]
@trws I haven't noticed any throughput issues with azure. I've been updating PRs with changes to develop and they seem to be going through.

**Review by rhornung67 — APPROVED** [review]

**Pull request merged**